add relative path solution to script's #load

nhaar opened the pull request:
## Description
If attempting to use `#load` inside a csx script (which allows you to load another script, useful for file organization), and you supply a relative path, it will not work as one might believe, that is, it will use the relative path to the UndertaleModTool directory and not a path relative to the script. To fix this, the goal is so that whenever someone uses #load to a relative path, it will "hardcode" that with an absolute path so that it will work.

### Caveats
I can't see anything troublesome with this. I have tested this in a project and it works perfectly

VladiStep commented on UndertaleModTool/MainWindow.xaml.cs:
```diff
@@ -2505,12 +2506,13 @@ public string ProcessException(in Exception exc, in string scriptText)
                     if (traceLine.TrimStart()[..13] == "at Submission") // only stack trace lines from the script
                     {
+                        // matches full path of the script file
+                        string sourceFile = Regex.Match(traceLine, @"(?<=in ).*\.csx(?=:line \d+)").Value;
```
```suggestion
                        string sourceFile = Regex.Match(traceLine, @"(?<=in ).*\.csx(?=:line \d+)").Value;
                        if (!File.Exists(sourceFile))
                            continue;
```

VladiStep commented on UndertaleModTool/MainWindow.xaml.cs:
```diff
@@ -2526,10 +2528,21 @@ public string ProcessException(in Exception exc, in string scriptText)
+                    string scriptPath = scriptLineNum.Item1;
+                    if (!scriptsCode.ContainsKey(scriptPath))
+                    {
+                        string scriptCode = File.ReadAllText(scriptPath);
```
```suggestion
                        string scriptCode = null;
                        try
                        {
                            scriptCode = File.ReadAllText(scriptPath);
                        }
                        catch (Exception e)
                        {
                            return $"An error occurred while processing the exception text.\nError message - \"{e.Message}\"\nThe unprocessed text is below.\n\n" + excString;
                        }
```

VladiStep commented on UndertaleModTool/MainWindow.xaml.cs:
```diff
@@ -2526,10 +2528,21 @@ public string ProcessException(in Exception exc, in string scriptText)
+                    }
+                }
+
+                string excLines = string.Join('\n', loadedScriptLineNums.Select(pair => $"Line {pair.Item2} in script {pair.Item1}: {scriptsCode[pair.Item1][pair.Item2 - 1]}")); // - 1 because line numbers start from 1
```
```suggestion
                string excLines = string.Join('\n', loadedScriptLineNums.Select(pair => 
                {
                    string scriptName = pair.Item1.Split(Path.DirectorySeparatorChar)[^1];
                    return $"Line {pair.Item2} in script {scriptName}: {scriptsCode[pair.Item1][pair.Item2 - 1]}"; // - 1 because line numbers start from 1
                }));
```

VladiStep reviewed (changes requested):
I just made it safer, and also made it write only script names instead of full path.

nhaar commented:
I have accepted VladiStep's changes

As for Miepee's request, I ran a test using `RunUMTScript` to see if it would work the same, unfortunately it doesn't (besides the fact that it doesn't accept path relatives to the script itself). One of the things I have been using `#load` for is that it allows to pass variables/functions defined in the script:

```cs
#load "script.csx"

SomeFunction(); // from the script
```

As annoying as it is that it doesn't allow for properly declaring what you want to get from the script, it is still useful, and the alternative with the builtin method doesn't work unfortunately.

Let me know if it would still be the goal to have `RunUMTScript` take relative paths through this PR.

Miepee commented:
> As annoying as it is that it doesn't allow for properly declaring what you want to get from the script

Huh. Makes me wonder what would be needed to change that, but that dosesn't sound easy at first glance. Guess using #load it is then.
I'll try to get to reviewing the PR then soon:tm:.

Miepee reviewed (commented):
no real review but a) you seem to have compilation errors and b) since you only change the gui, this will mean that it will not work on CLi

nhaar commented:
Something must have gone wrong after I accepted the commits, I will properly review for that later and move what needs to be on the CLI

nhaar commented:
I fixed the compilation issue and added it to the CLI

I wasnt sure where to put the method that adds the "workaround", so I made a new file. You can probably organize it better than me, if you don't like that

colinator27 commented:
As mentioned in the above PR I just linked, I looked further into this and found a solution that doesn't depend on regular expressions/replacing text in the script contents. That PR will probably supersede this one.
